### PR TITLE
ENH: Added labels to ophyd devices

### PR DESCRIPTION
**Pull request by jrmlhermitte**
WIP: From this issue [here](https://github.com/NSLS-II/bluesky/pull/1020).

### CODE REVIEW

File: ophyd/ophydobj.py [hunk 1]
```diff
@@ -33,8 +33,11 @@ class OphydObject:
 
     _default_sub = None
 
-    def __init__(self, *, name=None, parent=None):
+    def __init__(self, *, name=None, parent=None, labels=None):
         super().__init__()
+        if labels is None:
```
Comment (tacaswell) [inline ophyd/ophydobj.py]: I would move this above the super call so we can expect this to be available.

File: ophyd/ophydobj.py [hunk 2]
```diff
@@ -33,11 +33,11 @@ class OphydObject:
 
     _default_sub = None
 
-    def __init__(self, *, name=None, parent=None, device_labels=None):
+    def __init__(self, *, name=None, parent=None, labels=None):
```
Comment (mrakitin) [inline ophyd/ophydobj.py]: The changes here are opposite than the statement in the commit title. I'm a bit confused - do we eventually prefer `labels` over `device_labels`?

File: ophyd/ophydobj.py (hunk 1 above)
Comment (jrmlhermitte) [reply ophyd/ophydobj.py]: good idea thanks!

File: ophyd/ophydobj.py (hunk 2 above)
Comment (jrmlhermitte) [reply ophyd/ophydobj.py]: oops yes wrote too fast, went from `device_labels` to `lablels` will edit commit
thanks!

**Comment by jrmlhermitte** [comment]
The failures on Travis don't seem to be specifically related to this. It only fails py3.6 and nightly:

```
>       self.assertEqual(m.position, 0.1)
>       assert a == b
E       assert 0.099999999999999645 == 0.1
ophyd/tests/conftest.py:314: AssertionError
```

and
```

```

**Comment by jrmlhermitte** [comment]
@tacaswell do we have final blessing?
@klauer as well?

**Comment by klauer** [comment]
Overall 👍 

Is there a reason not to keep the attribute easily readable by the user? If not just as a public attribute, I'd think there should at least be a property. (These can be likened to hints, which are easily accessible by the user.)

**Comment by tacaswell** [comment]
We were channelling the IPython style `_repr_html_` naming scheme to indicate that this is part of the framework (which is something we have done no where else yet).

**Comment by danielballan** [comment]
Yeah. Two things:

1. I'm not opposed to a property that reflects this up as `.label` in a future release. Keeping it private for now makes it easier to undo this if it's has bad or unintended consequences.
2. This is different from other public bluesky/ophyd API because we're using it in the context of scraping a namespace _across libraries_. Underscore-named interfaces are one established way to do this, such as the way pandas talks IPython (and other consumers) via `_repr_html_`. A public attribute (`label`) or a private one with an unscoped name (`_label`) risks scooping unrelated objects with the same name that aren't trying to participate in this interface. And like `_repr_html_`, its intended use is UI/display-oriented.

Anyway, I think we should consider this an experimental feature.

**Comment by danielballan** [comment]
Taking in the discussion on related issues, I am hearing that people are generally buying that `_ophyd_labels_` is an acceptable experiment. I don't think we've established yet whether this information should be reflected in another, public attribute, and if so where. We have room to continue to discuss that while we try out this (via  https://github.com/NSLS-II/bluesky/pull/1020, https://github.com/NSLS-II/bluesky/pull/1031) at the beamlines and see how it goes.

But the related discussion around `*_attrs` prompted by #539 is very interesting and needs further discussion.

**Pull request merged**